### DISCUSSION TITLE
Welcome to XposedFakeLocation Discussions!

### DISCUSSION THREAD

**Discussion by noobexon1** [noobexon1/XposedFakeLocation#3] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by hhyk123** [comment]
微信为什么有时候用联网位置不变，或者改好了定位一会又变回原位置！

**Reply by noobexon1** [reply]
Please use English so that everyone can participate in the discussion :)

**Comment by hhyk123** [comment]
OK, now use this plug-in, WeChat. After changing the location, it will change back to the original location for a while, and then switch to the changed location, and then open the wireless network without changing the location. Can you add a base station customization function?



---Original---
From: ***@***.***&gt;
Date: Wed, Dec 25, 2024 22:56 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [noobexon1/XposedFakeLocation] Welcome to XposedFakeLocationDiscussions! (Discussion #3)




 
Please use English so that everyone can participate in the discussion :)

**Reply by noobexon1** [reply]
I guess adding network spoofing could be nice. It's also not that hard to do... I'm a bit busy with work lately, but the next version will include those features when it's out. Not sure when that would be though. cheers :)

**Reply by dautomobile007** [reply]
please add support to android 10 as well

**Reply by dautomobile007** [reply]
> please add support to android 10 as well

Update: I successfully converted it to run on Android 10. Thank you!

**Reply by dautomobile007** [reply]
> I guess adding network spoofing could be nice. It's also not that hard to do... I'm a bit busy with work lately, but the next version will include those features when it's out. Not sure when that would be though. cheers :)

Is it possible to produce RAW NMEA in future updates?

**Comment by moedwyn** [comment]
1. how to apply the module system wide?
2. What apps need to be enabled in Xposed for this module?

**Reply by aggrandize18-del** [reply]
To apply the XposedFakeLocation module system-wide (globally) using LSPosed, you need to configure the Module Scope within the LSPosed Manager. Since location data on Android is handled by both the system kernel and Google Play Services, simply "enabling" the module isn't enough; you must tell LSPosed which system processes to hook.
Follow these steps to apply it system-wide:
1. Install and Enable the Module
 * Download and install the XposedFakeLocation APK.
 * Open the LSPosed Manager app.
 * Go to the Modules tab (puzzle icon).
 * Tap on XposedFakeLocation and toggle the Enable switch.
2. Set the "System Wide" Scope
For a module to affect the entire device, it needs to hook into the core Android framework.
 * Inside the XposedFakeLocation module settings in LSPosed, look at the Scope section.
 * Required: Check System Framework (usually the first item in the list, package name android). This allows the module to intercept location calls at the system level.
 * Recommended (for apps like Google Maps/Uber): Check Google Play Services (package name com.google.android.gms). Most modern apps get their location via Play Services rather than the raw Android API.
 * (Optional) If there are specific apps you want to ensure are covered, you can check them individually (e.g., Chrome, Instagram), but selecting the "System Framework" and "Google Play Services" is generally what is meant by "system-wide."
3. Apply and Reboot
 * After selecting the scopes, reboot your device. LSPosed requires a reboot to properly hook into the System Framework.
 * Once the device restarts, open the XposedFakeLocation app itself.
 * Grant it the necessary permissions (Location and Notification).
 * Select a location on the map and tap the Start/Play button.
4. Verification
To ensure it is working system-wide:
 * Open Google Maps and see if your blue dot is at the spoofed location.
 * Open a browser (like Chrome) and search for "my location" or use a site like browserleaks.com/geo.
 * Note: Unlike traditional spoofing apps, you do not need to set this app as the "Mock Location App" in Android Developer Options. In fact, keeping that setting empty helps bypass detection in many apps.
Troubleshooting
 * Location not changing: Ensure you have disabled "Google Location Accuracy" (Wi-Fi/Bluetooth scanning) in your phone's system settings, as this can sometimes leak your real location via nearby Wi-Fi signals.
 * App detects spoofing: If an app still sees your real location, go back to LSPosed and ensure that the specific app is explicitly checked in the module's scope.
 * LSPosed status: Make sure the LSPosed notification says "LSPosed is loaded" or "Active." If it says "Disabled," check your Magisk/KernelSU settings to ensure Zygisk is enabled.
 *

**Reply by dautomobile007** [reply]
It is not working; we have to hook the target app along with the system framework, Android system, Google Play services, and Google service framework. We are requesting that the developer please explain how we can hook the entire system without hooking a specific target app, so that all apps that require location for working see the spoofed location from the system itself, without hooking that specific app.
